docs/site/DEVELOPING.md

```diff
@@ -126,6 +127,46 @@ npm script `lint:fix`.
 npm run lint:fix
 ```
 
+## Working with dependencies
+
+We use npm's
+[package-lock feature](https://docs.npmjs.com/files/package-lock.json) to speed
+up our development workflow and CI builds.
+
+When installing dependencies in individual packages, `lerna bootstrap` calls
+`npm ci` to install (deep) dependencies as specified in `package-lock.json`
+file.
+
+Top-level dependencies are installed either from `package-lock.json` (when you
+run `npm ci`), or resolved freshly from the npm registry (when you run
+`npm install`).
+
+**IMPORTANT: Dependencies resolved locally within the monorepo must be excluded
+from package-lock files.**
+
+### Fixing package locks
+
+If you ever end up with corrupted package locks, run the following commands to
+fix the problem:
+
+```
+$ npx lerna clean && npx lerna bootstrap --no-ci
+```
+
+### Adding or updating dependencies
+
+Use the following command to add or update dependency `dep` in a package `name`:
+
+```
+$ npx lerna add --scope ${name} ${dep}
```

Comment:
I guess we will want to rework this section too, to explain that besides `lerna add`, dependencies can be also added directly by modifying `package.json` files and running `npm i` to update package locks. WDYT?

Reply:
I agree. Maybe you can remove this section and I'll add it as part of the follow-up PR to fix the `update-package-lock-json` script.